File: 342/power-of-four.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isPowerOfFour = function (num) {
    if (num == 0)
        return false;
    let a = ~~(Math.log(num) / Math.log(4));
    return num === 1 << a * 2;
};

File: 342/power-of-four.go
package main

import "fmt"

func isPowerOfFour(n int) bool {
	if n < 1 {
		return false
	}

	for n%4 == 0 {
		n /= 4
	}

	return n == 1
}

func main() {
	n := 16
	fmt.Println(isPowerOfFour(n))
}
